### ISSUE TITLE
Stack overflow in KVUtils

### ISSUE THREAD

**Issue by bame-da** [digital-asset/daml#5947]
The below DAML Model creates a stack overflow in kvutils if `CreateN` is called with a large count. Can be tested using the attached `daml.yaml` by running `daml start` in SDK 1.0.

This works in Sandbox Classic so I believe it's a kvutils issue, not an engine issue.

DAML:
```
daml 1.2
module Batch where
import Daml.Script
template B
  with
    p : Party
  where
    signatory p
    nonconsuming choice CreateN : () 
      with
        n : Int
      controller p
      do
        res <- forA [1..n] (\_ -> do
              create T with ..
          )
        return()
template T
  with
    p : Party
  where
    signatory p
t : Script () = do
  p <- allocateParty "p"
  b <- submit p do createCmd B with ..
  submit p do exerciseCmd b CreateN with n = 5000
  return ()  
```

daml.yaml:
```
sdk-version: 1.0.0
name: batch
source: daml
parties:
  - P
version: 0.0.1
init-script: Batch:t
dependencies:
  - daml-prim
  - daml-stdlib
  - daml-script
sandbox-options:
  - --wall-clock-time
```

stackoverflow:
```
...
	at com.daml.ledger.participant.state.kvutils.committing.Common$Commit.$anonfun$flatMap$1(Common.scala:43)
	at com.daml.ledger.participant.state.kvutils.committing.Common$Commit.$anonfun$flatMap$1(Common.scala:43)
	at com.daml.ledger.participant.state.kvutils.committing.Common$Commit.$anonfun$flatMap$1(Common.scala:43)
...
```

**Comment by fabiotudone-da** [comment]
This should be fixed by #5901 but the self-contained DAML code that reproduces it should allow us to write an actual test case for it, which we should do.

**Comment by bame-da** [comment]
I can confirm that this is fixed in HEAD.
